parsons/databases/table.py

```diff
@@ -214,18 +214,17 @@ def dedup_table(
         partition = ", ".join(columns_list)
 
         dedup_query = f"""
-            alter table {self.table}
-            rename to {self.table}_temp_{current_timestamp};
-            create table {self.table} as
-            select * from
+            create table {self.table}_temp_{current_timestamp} as
             (select *
             , row_number() over (partition by {partition}
             order by {order_by_column_name} {order_by_direction}) as dup
-            from {self.table}_temp_{current_timestamp})
+            from {self.table})
             where dup=1;
-            alter table {self.table}
+            alter table {self.table}_temp_{current_timestamp}
             drop column dup;
-            drop table {self.table}temp_{current_timestamp} {run_cascade};
+            truncate table {self.table}
+            insert into {self.table} (select * from {self.table}_temp_{current_timestamp})
+             {run_cascade};
```

Comment:
Apologies as my SQL is rusty, but do we need to drop the temp table after copying it to the main table?

Reply:
Good point!